zwave_js_server/model/node/__init__.py

```diff
@@ -417,30 +426,25 @@ def update(self, data: NodeDataType) -> None:
                 # If we can't parse the value, don't store it
                 pass
 
-        new_endpoints_data = {
-            endpoint["index"]: endpoint for endpoint in self.data.pop("endpoints")
-        }
-        new_endpoint_idxs = set(new_endpoints_data)
-        stale_endpoint_idxs = set(self.endpoints) - new_endpoint_idxs
+    def update(self, data: NodeDataType) -> None:
+        """Update the internal state data."""
+        self.data = copy.deepcopy(data)
+        self._device_config = DeviceConfig(self.data.get("deviceConfig", {}))
+        if (device_class := self.data.get("deviceClass")) is None:
+            self._device_class = None
+        else:
+            self._device_class = DeviceClass(device_class)
 
-        # Remove stale endpoints
-        for endpoint_idx in stale_endpoint_idxs:
-            self.endpoints.pop(endpoint_idx)
+        self._statistics = NodeStatistics(
+            self.client, self.data.get("statistics", DEFAULT_NODE_STATISTICS)
+        )
+        if last_seen := data.get("lastSeen"):
+            self._last_seen = datetime.fromisoformat(last_seen)
+        if not self._statistics.last_seen:
```

Comment:
Does this mean that last seen in the statistics can be different than the last seen of the latest data?

Reply:
no it means that the statistics last seen can be missing in which case we populate it to keep everything in sync